src/components/button.tsx

```diff
@@ -41,3 +42,25 @@ export const Button = (props: any) => {
     </TouchableOpacity>
   )
 }
+
+export const ButtonIcon = (props: any) => {
+  return (
+    <TouchableOpacity
+      {...props}
+      style={[
+        {
+          width: 44,
+          height: 44,
+          ...props.style,
```

Comment:
I usually put the spread at the bottom of the object, just so everything can be overridden if need be

Reply:
But I'd expect props passed in to take priority. In the above if I pass in width: 50 in the props then the component will end up width 50. I think you're doing it the wrong way around?

Reply:
No, it's the exact same as you. You've got the `...props.style` underneath your width and height so your example confuses me. But how you've done it, alignItems and justifyContent are both 'center' and they can never be overriden because this spread is in the middle of the object